File: android/src/main/java/com/fastsecurestorage/FastSecureStoragePackage.java
package com.fastsecurestorage;

import java.util.Map;
import java.util.HashMap;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.bridge.NativeModule;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.TurboReactPackage;
import com.facebook.react.module.model.ReactModuleInfo;
import com.facebook.react.module.model.ReactModuleInfoProvider;

public class FastSecureStoragePackage extends TurboReactPackage {

  @Nullable
  @Override
  public NativeModule getModule(String name, @NonNull ReactApplicationContext reactContext) {
    if (name.equals(FastSecureStorageModule.NAME)) {
      return new FastSecureStorageModule(reactContext);
    } else {
      return null;
    }
  }

  @Override
  public ReactModuleInfoProvider getReactModuleInfoProvider() {
    return () -> {
      final Map<String, ReactModuleInfo> moduleInfos = new HashMap<>();
      moduleInfos.put(
        FastSecureStorageModule.NAME,
        new ReactModuleInfo(
          FastSecureStorageModule.NAME,
          FastSecureStorageModule.NAME,
          false, // canOverrideExistingModule
          false, // needsEagerInit
          true, // hasConstants
          false, // isCxxModule
          true // isTurboModule
        ));
      return moduleInfos;
    };
  }
}

File: android/src/main/java/com/fastsecurestorage/FastSecureStoragePackage.kt
package com.fastsecurestorage

import com.facebook.react.TurboReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.module.model.ReactModuleInfo
import com.facebook.react.module.model.ReactModuleInfoProvider
import com.fastsecurestorage.FastSecureStorageModule

class FastSecureStoragePackage : TurboReactPackage() {
  override fun getModule(
    name: String,
    reactContext: ReactApplicationContext,
  ): NativeModule? =
    if (name == FastSecureStorageImpl.NAME) {
      FastSecureStorageModule(reactContext)
    } else {
      null
    }

  override fun getReactModuleInfoProvider(): ReactModuleInfoProvider =
    ReactModuleInfoProvider {
      val moduleInfos: MutableMap<String, ReactModuleInfo> =
        HashMap()
      moduleInfos[FastSecureStorageImpl.NAME] =
        ReactModuleInfo(
          FastSecureStorageImpl.NAME,
          FastSecureStorageImpl.NAME,
          false, // canOverrideExistingModule
          false, // needsEagerInit
          true, // hasConstants
          false, // isCxxModule
          true, // isTurboModule
        )
      moduleInfos
    }
}
